### ISSUE TITLE
gitlab-ci: handle npm depend on vendor

### ISSUE THREAD

**Issue by simonbuehler** [lorisleiva/laravel-deployer#241]
hi,

using the 'old' package in gitlab, but should also be relevant for new:

for e.g. tailwind css treeshaking with postCSS npm mix needs to scan also vendor for used css classes, i can only make npm build script depend on composer output when npm is in anotehr stage , so composer is in build, npm in test but this looks fishy.

whats the correct way for handling this?

sample tailwind.config.js extract:
```
module.exports = {
    purge: {
        content: [
            './vendor/laravel/framework/src/Illuminate/Pagination/resources/views/*.blade.php',
            './vendor/laravel/jetstream/**/*.blade.php',
            './storage/framework/views/*.php',
            './resources/views/**/*.blade.php',
            // Tall-forms start
            './config/tall-forms.php',
            './vendor/tanthammar/tall-forms/**/*.php',
            './vendor/tanthammar/tall-forms-sponsors/**/*.php',
            // Tall-forms end
            './vendor/rappasoft/laravel-livewire-tables/**/*.php'
        ],
        options: {
            defaultExtractor: (content) => content.match(
                /[^<>"'`\s]*[^<>"'`\s:]/g) || [],
            safeList: [/-active$/, /-enter$/, /-leave-to$/, /show$/],
        },
    },
....
```

**Comment by lorisleiva** [comment]
Hi there 👋 

Sorry but I'm not sure what your question is and how it has something to do with Laravel Deployer.

**Comment by simonbuehler** [comment]
ok let me rephrase the issue:

in gitlab ci npm depends on the output of composer to shake css tree also for vendor content; the only way to provide npm with composer stage output is in another stage. 

following https://lorisleiva.com/laravel-deployment-using-gitlab-pipelines , is it needed to add another stage in gitlab or whats the best practice for this case?

**Comment by lorisleiva** [comment]
Oh I see. Not it's absolutely not needed, you may run them in the same job. The bonus of having them in separate jobs is that they can run in parallel making the pipeline slightly faster but if one depends on the other than you've got no choice. 🙂

**Comment by simonbuehler** [comment]
actually its no a deployer issue but rather a gitlab ci, closing but still interested in a hint!
